ds3-autogen-api/src/main/java/com/spectralogic/ds3autogen/api/models/apispec/Ds3Element.java

```diff
@@ -81,12 +81,14 @@ public boolean equals(final Object obj) {
             return false;
         }
         final Ds3Element element = (Ds3Element) obj;
-        return Objects.equals(this.getName(), element.getName())
-                && Objects.equals(this.getType(), element.getType())
-                && Objects.equals(this.getComponentType(), element.getComponentType());
+
+        return this.getName().equals(element.getName())
+                && this.getType() != null
+                && this.getType().equals(element.getType())
+                && this.getComponentType() != null
```

Comment:
Name, type, and component type are parsed directly from the contract. Of these, only component type is expected to be null sometimes.  Name is not derived.